README.md

```diff
@@ -427,6 +428,50 @@ func main() {
     client.Close()
 }
 ```
+
+### SQL
+
+InfluxDB Cloud now supports the ability to query with SQL. With this support
+a user can pass in an SQL query to get a Arrow reader and iterate through the
+records.
+
+> :warning: Parameterized Queries are supported only in InfluxDB Cloud. There is no support in InfluxDB OSS currently.
+
+```go
+package main
+
+import (
+	"context"
+	"fmt"
+
+	influxdb2 "github.com/influxdata/influxdb-client-go/v2"
+)
+
+func main() {
+	client := influxdb2.NewClient("http://us-east-1-1.aws.cloud2.influxdata.com:8086", "my-token")
+    defer client.Close()
+
+	querySQLAPI := client.QuerySQLAPI()
+	result, err := querySQLAPI.Query(context.Background(), "my-bucket", "select * from cpu limit 10")
+	if err != nil {
+		panic(err)
+	}
+
+    for {
+		if nextRecord, err := reader.Read(); err == io.EOF {
+			reader.Release()
+			break
+		} else if err != nil {
+			panic(fmt.Errorf("unable to read record: %v", err))
+		} else {
+			fmt.Print(nextRecord)
+		}
+	}
```

Comment:
I believe this is completely wrong. The general Go idiom for iterating over result set is using `obj.Next()` and check for an error via `obj.Err()`, an the retuned reader supports it:
```go
    defer result.Release()
    for result.Next() {
        fmt.Println(result.Record())
    }
    if result.Err() != nil {
        panic(fmt.Errorf("unable to read record: %v", err))
    }
```

Reply:
Also, such an example needs to be added to API docs examples in `api/examples_test.go`. 
Note: to overcome the circular import issue there is a fake client that needs to be enhanced with the QuerySQLAPI() function.
```go
func ExampleQuerySQLAPI() {
	client := influxdb2.NewClient("http://us-east-1-1.aws.cloud2.influxdata.com:8086", "my-token")
	defer client.Close()
	querySQLAPI := client.QuerySQLAPI()
	result, err := querySQLAPI.Query(context.Background(), "my-bucket", "select * from cpu limit 10")
	if err != nil {
		panic(err)
	}
	defer result.Release()
	for result.Next() {
		fmt.Println(result.Record())
	}
	if result.Err() != nil {
		panic(fmt.Errorf("unable to read record: %v", err))
	}
}
```